src/server/svr_recov_db.c

```diff
@@ -477,6 +477,15 @@ sched_save_db(pbs_sched *ps, int mode)
 	obj.pbs_db_obj_type = PBS_DB_SCHED;
 	obj.pbs_db_un.pbs_db_sched = &dbsched;
 
+	/*
+	 * default values for scheduler object
+	 */
+	if ((ps->sch_attr[(int)SCHED_ATR_sched_cycle_len].at_flags & ATR_VFLAG_SET) == 0) {
```

Comment:
There are two usecases where we need to set default attributes 

1) While doing attribute unset
2) While creating sched objects. 
sched_alloc is called during creating sched object. So cannot be part of  sched_alloc function